cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -80,6 +81,13 @@ type CMServer struct {
 	ServiceAccountKeyFile   string
 	RootCAFile              string
 
+	// volumeConfig
+	PersistentVolumeRecyclerDefaultScrubPod          string
+	PersistentVolumeRecyclerMinTimeoutNfs            int
```

Comment:
This is the biggest question of this PR:   strongly typed struct containing all config values or a generic map of values?

typed struct:   strongly typed like everything else in Go.  requires all volume configurable properties  be maintained in the VolumeConfig struct.  CLI flags are 1:1 with properties on VolumeConfig.

map config:  not strongly typed, all string values opaque to all but the plugin using it.  How do we know which CLI flags get put into the map?  Use a prefix as convention?  Do we validate all CLI flags when we initialize volumes and die on an invalid value?

If adding a CLI flag maps 1:1 to some piece of config, does it make more or less sense to get the type correctly set on VolumeConfig?

Considering how explicitly everything else is strongly typed, VolumeConfig as a struct makes sense to me.  Each plugin's configurable properties would still need to be added to the CLI explicitly, making a generic map seemingly less useful.  If we're adding to the CLI, we might as well bind that value to VolumeConfig and have it typed.   My $0.02.

Everything else is straightforward if we get this fundamental question answered first.